File: Java/src/storm/xmlbinder/binder/AbstractCollectionBinder.java
package storm.xmlbinder.binder;

import java.util.Collection;

import storm.xmlbinder.XmlBinderException;

/**
 * Abstract class for all CollectionBinder class.
 * Provide some useful method.
 * @author Storm <j-storm@hotmail.fr>
 */
public abstract class AbstractCollectionBinder extends AbstractBinder 
{
	/**
	 * Constructor to initialize super class with the field name.
	 * @param _fieldName : the name of the field.
	 */
	protected AbstractCollectionBinder(String _fieldName) 
	{
		super(_fieldName);
	}
	
	/**
	 * Method to add an element to the specified collection field. 
	 * @param _object : the object to add to the collection.
	 * @param _data : the data object.
	 * @param _fieldName : the name of the collection field. 
	 * @throws XmlBinderException
	 */
	protected void add(Object _object, Object _data, String _fieldName) throws XmlBinderException
	{
		@SuppressWarnings("unchecked")
		Collection<Object> collection = (Collection<Object>) getField(_data, _fieldName);
		collection.add(_object);
	}
}

File: C#/Abstract/AbstractCollectionBinder.cs
using System;
using System.Collections;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace XmlBinder.Abstract
{
	/// <summary>
	/// Abstract class for all collection binder item.
	/// </summary>
	public abstract class AbstractCollectionBinder : AbstractBinder
	{
		/// <summary>
		/// Create a new CollectionBinder with specified field name.
		/// </summary>
		/// <param name="_fieldName">The name of the collection field.</param>
		public AbstractCollectionBinder(string _fieldName)
			: base(_fieldName)
		{

		}

		/// <summary>
		/// Method to add an element to a collection.
		/// </summary>
		/// <param name="_source">Source object.</param>
		/// <param name="_fieldName">Name of the collection field in source object.</param>
		/// <param name="_add">The object to add.</param>
		public void addElement(object _source, string _fieldName, object _add)
		{
			IList collection = (IList)Tools.FieldAccess.getField(_source, _fieldName);
			collection.Add(_add);
		}


	}
}
